### PR TITLE
Use PyTorch's `any` implementation

### PR DESCRIPTION
**Pull request by alanhdu**
Python's `any` is much slower than PyTorch's `any` (especially if the data lives on the GPU and you can avoid unnecessary GPU -> CPU copy of `nans`), so this will be much faster.

### CODE REVIEW

**Comment by Borda** [comment]
@alanhdu interesting, mind include some benchmark? :rabbit:

**Comment by alanhdu** [comment]
The effect's pretty obvious -- for example, I did:
```python
sm = SumMetric()
X = torch.randn(16, 200, 10)
```
and then
```
%%time
for __ in range(50):
    sm.update(X)
print(sm.compute())
```

Before, this took 3.8 seconds to run on my laptop -- afterwards, this takes 6.31 ms (a 500x speedup). I assume this would be even more drastic if this was running on a GPU.

**Comment by justusschock** [comment]
Which makes sense, as python's any requires boolean values and therefore needs to cast every tensor element to a boolean (which includes a potential GPU->CPU transfer and thereby a synchronization point.

**Review by Borda — APPROVED** [review]

File: torchmetrics/aggregation.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def _cast_and_nan_check_input(self, x: Union[float, Tensor]) -> Tensor:
             x = torch.as_tensor(x, dtype=torch.float32, device=self.device)
 
         nans = torch.isnan(x)
-        if any(nans.flatten()):
+        if nans.any().item():
```
Comment (justusschock) [inline torchmetrics/aggregation.py:72]:
```suggestion
        if nans.any():
```

**Pull request merged**